sql/core/src/main/scala/org/apache/spark/sql/Column.scala

```diff
@@ -68,6 +68,25 @@ class TypedColumn[-T, U](
     }
     new TypedColumn[T, U](newExpr, encoder)
   }
+
+  /** Creates a TypedColumn based on the given expression. */
+  private def withExpr(newExpr: Expression): TypedColumn[T, U] =
+    new TypedColumn[T, U](newExpr, encoder)
+
+  /**
+   * Gives the TypedColumn a name (alias).
+   * If the current TypedColumn has metadata associated with it, this metadata will be propagated
+   * to the new column.
+   *
+   * @group expr_ops
+   * @since 2.0.0
+   */
+  override def as(alias: String): TypedColumn[T, U] = withExpr {
```

Comment:
@rxin @cloud-fan : Thanks very much. I have made the changes based on your comments. Can you help check?